experiments/Marmot/BezierBoundingBox/BezierBoundingBox.tex

```diff
@@ -0,0 +1,173 @@
+following discussion uses in part the results from
+\texttt{https://pomax.github.io/bezierinfo/}. What is a cubic Bezier curve? A
+cubic Bezier curve running from $(x_0,y_0)$ to $(x_1,y_1)$ with control points
+$(x_a,y_a)$ and $(x_a,y_a)$ can be parametrized by
+\begin{equation}
+ \gamma(t)~=~
+ \begin{pmatrix} x(t)\\ y(t) \end{pmatrix}~=~
+ \begin{pmatrix}t^3 x_{1}+3 t^2 (1-t) x_{b}+(1-t)^3
+   x_{0}+3 t (1-t)^2 x_{a}\\
+   t^3 y_{1}+3
+   t^2 (1-t) y_{b}+(1-t)^3 y_{0}+3 t (1-t)^2
+   y_{a}\end{pmatrix}\;,
+\end{equation}
+where $t$ runs from 0 to 1 (and $\gamma(0)=(x_0,y_0)$ and
+$\gamma(1)=(x_1,y_1)$). Surely, the bounding box has to contain
+$(x_0,y_0)$ and $(x_1,y_1)$. If the functions $x(t)$ and $y(t)$ have extrema in
+the interval $[0,1]$, then the bounding box will in general be larger than that.
+In order to determine the extrema of the curve, all
+we need to find the extrema of the functions $x(t)$ and $y(t)$ for $0\le t\le
+1$. That is, we need to find the solutions of the quadratic equations
+\begin{equation}
+ \frac{\mathrm{d}x}{\mathrm{d}t}(t)~=~0\quad\text{and}\quad
+ \frac{\mathrm{d}y}{\mathrm{d}t}(t)~=~0\;.
+\end{equation}
+Let's discuss $x$, $y$ is analogous. If the discriminant
+\begin{equation}
+ d~:=~x_{0} x_{1}-x_{0}
+   x_{b}-x_{1}
+   x_{a}+x_{a}^2-x_{a}
+   x_{b}+x_{b}^2
+\end{equation}
+is greater than 0, there are two solutions
+\begin{equation}
+ t_\pm~=~\frac{x_{0}-2
+   x_{a}+x_{b}\pm\sqrt{x_{0} x_{1}-x_{0}
+   x_{b}-x_{1}
+   x_{a}+x_{a}^2-x_{a}
+   x_{b}+x_{b}^2}}{x_{0}-x_{1}-3
+   x_{a}+3 x_{b}} \;.
+\end{equation}   
+In this case, we need to make sure that the bounding box contains, say
+$(x(t_-),y_0)$ and $(x(t_+),y_0)$. If $d\le0$, the bounding box does not need to
+be increased in the $x$ direction. As already mentioned, the analogous
+statements apply to $y(t)$. 
+
+It is rather straightforward to implement this
+prodecure in Ti\emph{k}Z. The relevant macros are \verb|\pgf@lt@curveto| (and
+\verb|\pgf@nlt@curveto|).\footnote{Some care has to be taken when squaring
+lengths since they are all measured in points, and the square can easily become
+large and trigger a \texttt{dimension too large} error. When computing the
+discriminant $d$, I thus divided these distances by some taming factor that I
+took to be 32 for no special reasons. It might very well be that there are
+better taming factors, or that one needs to dial the taming factor as a function
+of the input values.} The macro \verb|\pgf@lt@curveto| takes six arguments,
+which are $x_a$, $y_a$, $x_b$, $y_b$, $x_1$ and $y_1$ (in that order). $x_0$ and
+$y_0$ are stored in \verb|\pgf@path@lastx| and \verb|\pgf@path@lastx|,
+respectively.
+
+\subsection*{Examples}
+
+\makeatletter
+\def\pgf@lt@curveto#1#2#3#4#5#6{%
+  % extrema in x
+  \pgfmathparse{((#1/32)*(#1/32)-1*((#1/32)*(#3/32))+(#3/32)*(#3/32)-1*((#1/32)*(#5/32))+(-(#3/32)+(#5/32))*(\pgf@path@lastx/32))}%
+  \pgfutil@tempdima=\pgfmathresult pt % <- why do I need this space?
```

Comment:
You need this space, because otherwise TeX keeps looking for a unit and expands tokens.